### PR TITLE
introduce xxhash into third_party

### PR DESCRIPTION
**Pull request by fzakaria**
Introduce xxhash the Extremely fast non-cryptographic hash algorithm
https://github.com/Cyan4973/xxHash
commit 5b45185d0419f63eb5911133f3dc578a163d162f

```console
 ./o/third_party/xxhash/xxhsum.com -h
xxhsum.com 0.8.1 by Yann Collet 
Print or verify checksums using fast non-cryptographic algorithm xxHash 

Usage: xxhsum.com [options] [files] 

When no filename provided or when '-' is provided, uses stdin as input. 
```

### CODE REVIEW

File: third_party/xxhash/cli/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+// clang-format off
```
Comment (fzakaria) [inline third_party/xxhash/cli/README.md:1]: the python third-party script included these clang-format off to every file

File: third_party/xxhash/cli/xsum_config.h [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+#endif
+
+/**
+ * 
+ * fzakaria: remove this
+ * 
+ * #if !defined(S_ISREG)
+ * #  define S_ISREG(x) (((x) & S_IFMT) == S_IFREG)
+ * #endif
+ */
```
Comment (fzakaria) [inline third_party/xxhash/cli/xsum_config.h:113]: I had to hand-edit the source here -- this macro becomes redefined by the libc the way the includes work.

File: third_party/xxhash/tests/Makefile [hunk 3]
```diff
@@ -0,0 +1,132 @@
+// clang-format off
```
Comment (fzakaria) [inline third_party/xxhash/tests/Makefile:1]: TODO: do we want to support tests in third_party?

**Review by jart — CHANGES_REQUESTED** [review]
Could you create a README.cosmo file describing this package including where it came from? Also if it wouldn't be too much trouble, could you make this pull request consist of two commits: (1) the first commit should basically be copy/pasted upstream sources verbatim, and (2) the second commit should be what you have now. That way it's clearer where the third party code starts and where our local modifications begin.

**Comment by fzakaria** [comment]
I noticed lz4cli also uses xxhash but has the source within itself -- that can be fixed after this.

File: third_party/xxhash/tests/Makefile (hunk 3 above)
Comment (jart) [reply third_party/xxhash/tests/Makefile:1]: Generally speaking, yes. It'd be very much welcome if you went to the extra effort to do that.

**Comment by fzakaria** [comment]
Added the README.cosmo.

Trying to figure out the cleanest `git rebase` to introduce the commit earlier as you mentioned.

**Review by jart — APPROVED** [review]
Let's make progress on getting Mold into Cosmopolitan, now that you've got the process nailed down with zstd. Thanks for the contribution!

**Pull request merged**